### PR TITLE
feat(pipelines): add support for Tekton Sidecar containers

### LINKED ISSUE #7488: Support Tekton Sidecars in JX Pipeline syntax
### Summary 

Tekton has native support for [sidecar containers](https://github.com/tektoncd/pipeline/blob/v0.11.3/docs/taskruns.md#sidecars). A good use-case for sidecars in build pipelines is to run integration tests using Selenium: we need to add both selenium container and one or more browser containers in the build pod.

For the moment we have a workaround by using a custom mutating admission webhook controller, to inject sidecar containers in the build pods at creation time. But it would be easier to specify them in the pipeline of course.

### PR DESCRIPTION
**Pull request by vbehar**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

Adding support for [Tekton Sidecars](https://github.com/tektoncd/pipeline/blob/v0.11.3/docs/taskruns.md#sidecars) at the pipeline and/or stage level - in the options:

```
pipelineConfig:
  pipelines:
    release:
      pipeline:
        options:
          sidecars:
            - name: top-level-sidecar
              image: top-level/sidecar:tag
        stages:
          - name: my-stage
            options:
              sidecars:
                - name: stage-level-sidecar
                  image: stage-level/sidecar:latest
```

#### Special notes for the reviewer(s)

- Note that as of Tekton 0.11.3 (current version of Tekton used by jx), sidecars have a few known bugs: https://github.com/tektoncd/pipeline/blob/v0.11.3/docs/taskruns.md#sidecars
- PR to update the docs: https://github.com/jenkins-x/jx-docs/pull/3001

#### Which issue this PR fixes

fixes #7488

### CODE REVIEW

**Comment by vbehar** [comment]
/assign @abayer 
when you have time ;-) thanks!

File: pkg/tekton/syntax/pipeline_test.go [hunk 1]
```diff
@@ -1235,6 +1235,84 @@ func TestParseJenkinsfileYaml(t *testing.T) {
...
+						),
+					),
+					sh.StageStep(
+						sh.StepCmd("echo"),
+						sh.StepArg("hello"), sh.StepArg("world"),
+						sh.StepName("A Step With Spaces And Such"),
+					),
+				),
+			),
+			pipeline: tb.Pipeline("somepipeline-1", "jx", tb.PipelineSpec(
```
Comment (abayer) [inline pkg/tekton/syntax/pipeline_test.go:1277]: fyi, now that #7487 has merged, this'll need to change to `tb.Pipeline("somepipeline-1", tb.PipelineNamespace("jx"), ...`, and `tb.Task(...)` will need to change to `tb.Task("somepipeline-a-working-stage-1", tb.TaskNamespace("jx"), ...`.

Comment (vbehar) [reply pkg/tekton/syntax/pipeline_test.go]: thanks, I rebased and fixed it

**Comment by rawlingsj** [comment]
This looks good to me but @abayer is best to approve from here.

**Pull request merged**